File: mywebsite-parent/mywebsite-manager-web/src/main/webapp/angularjs/js/service/adminService.js
app.service('adminService', function ($http) {

    /**
     * 获取登录信息
     * @returns {*}
     */
    this.getLoginInfo = function () {
        return $http.get('../admin/getLoginInfo.do');
    };

    /**
     * 根据email查询用户
     * @param email
     * @returns {*}
     */
    this.findOneByEmail = function (email) {
        return $http.get('../admin/findOneByEmail.do?email='+ email);
    };

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param pageSize
     * @param pageNum
     * @returns {*}
     */
    this.search = function (searchEntity, pageSize, pageNum) {
        return $http.post('../admin/search.do?rows=' + pageSize + '&page=' + pageNum, searchEntity);
    };

    /**
     * 修改用户信息
     * @param updateEntity
     * @returns {*}
     */
    this.update = function (updateEntity) {
        return $http.post('../admin/update.do', updateEntity);
    };

    /**
     * 删除用户
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../admin/deleteIds.do?ids=' + ids);
    };

    /**
     * 批量审核
     * @param ids
     * @param status
     * @returns {*}
     */
    this.updateStatusIds = function (ids, status) {
        return $http.get('../admin/updateStatusIds.do?ids=' + ids + '&status=' + status);
    };

    /**
     * 发送邮箱验证码
     * @param id
     * @returns {*}
     */
    this.sendPasswordEmail = function (id) {
        return $http.get('/admin/sendPasswordEmail.do?id=' + id);
    };

    /**
     * 更新用户密码
     * @param entity
     * @returns {*}
     */
    this.updatePassword = function (entity) {
        return $http.post('../admin/updatePassword.do', entity);
    };

    /**
     * 更新邮箱
     * @param entity
     * @returns {*}
     */
    this.updateEmail = function (entity) {
        return $http.post('../admin/updateEmail.do', entity);
    };

    /**
     * 发送邮箱验证码
     * @param entity
     * @returns {*}
     */
    this.sendReEmail = function (entity) {
        return $http.post('/admin/sendReEmail.do',entity);
    };

    /**
     * 登出
     * @returns {*}
     */
    this.logout = function () {
        return $http.get("/logout");
    };
});

File: mywebsite-parent-v2.0/mywebsite-manager-web/src/main/java/xyz/lirui123/mywebsite/manager/service/AdminService.java
package xyz.lirui123.mywebsite.manager.service;

import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.stereotype.Component;
import xyz.lirui123.mywebsite.pojo.TbAdmin;
import xyz.lirui123.mywebsite.response.ResponseResult;

public interface AdminService extends UserDetailsService {

    /**
     * 根据用户名查询用户
     *
     * @param admin
     * @return
     */
    public abstract TbAdmin findOneByAdmin(String admin);

    /**
     * 根据Id查询用户
     *
     * @param id
     * @return
     */
    public abstract TbAdmin findOneById(Long id);

    /**
     * 根据邮箱查询用户
     *
     * @param email
     * @return
     */
    public abstract TbAdmin findOneByEmail(String email);


    /**
     * 添加用户
     *
     * @param admin
     */
    public abstract ResponseResult add(TbAdmin admin);

    /**
     * 账号激活
     *
     * @param id
     * @param token
     * @return
     */
    public abstract ResponseResult active(Long id, String token);

    /**
     * 条件分页查询
     *
     * @param admin
     * @param pageSize
     * @param pageNum
     * @return
     */
    public abstract ResponseResult search(TbAdmin admin, int pageSize, int pageNum);

    /**
     * 修改用户信息
     *
     * @param admin
     * @return
     */
    public abstract ResponseResult update(TbAdmin admin);

    /**
     * 删除用户
     * @param ids
     * @return
     */
    public abstract ResponseResult delete(Long[] ids);

    /**
     * 审核用户
     * @param ids
     * @param status
     * @return
     */
    public abstract ResponseResult updateStatusIds(Long[] ids, String status);

    /**
     * 修改密码
     * @param admin
     * @return
     */
    public abstract ResponseResult updatePassword(TbAdmin admin);

    /**
     * 发送密码验证邮箱
     * @param id
     * @return
     */
    public abstract ResponseResult sendPasswordEmail(Long id);

    /**
     * 修改用户邮箱
     * @param admin
     * @return
     */
    public abstract ResponseResult updateEmail(TbAdmin admin);

    /**
     * 发送邮箱绑定验证邮箱
     * @param admin
     * @return
     */
    public abstract ResponseResult sendReEmail(TbAdmin admin);
}

